File: src/main/java/com/example/service/ItemService.java
package com.example.service;

import com.example.dto.item.ItemCreationDto;
import com.example.dto.item.ItemDto;
import com.example.dto.item.ItemUpdateDto;

import java.util.List;
import java.util.UUID;

public interface ItemService {
    List<ItemDto> findAll();
    ItemDto findById(UUID uuid);
    ItemDto save(UUID categoryId, ItemCreationDto creationDto);
    ItemDto updateItem(UUID itemId, ItemUpdateDto updateDto);
    void deleteById(UUID uuid);
}

File: src/main/java/com/example/service/ItemService.kt
package com.example.service

import com.example.dto.image.ImageCreationDto
import com.example.dto.item.ItemCreationDto
import com.example.dto.item.ItemDto
import com.example.dto.item.ItemUpdateDto
import java.util.UUID

interface ItemService {
    fun findById(id: UUID): ItemDto
    fun findAll(): List<ItemDto>
    // todo fun filter(filterDto: ItemFilterDto): List<ItemDto>
    fun save(creationDto: ItemCreationDto): ItemDto
    fun updateItem(id: UUID, updateDto: ItemUpdateDto): ItemDto
    fun setImage(id: UUID, imageDto: ImageCreationDto): ItemDto
    fun deleteById(id: UUID)
}